internal/receiver/smartagentreceiver/receiver_test.go

```diff
@@ -67,6 +83,7 @@ func newConfig(nameVal, monitorType string, intervalSeconds int) Config {
 }
 
 func TestSmartAgentReceiver(t *testing.T) {
+	t.Cleanup(cleanUp)
```

Comment:
Not requesting, but noting testify provides fixtures in its suite helper: https://pkg.go.dev/github.com/stretchr/testify/suite#BeforeTest